homeassistant/components/zimi/config_flow.py

```diff
@@ -0,0 +1,142 @@
+"""Config flow for zcc integration."""
+
+from __future__ import annotations
+
+import logging
+import socket
+from typing import Any
+
+import voluptuous as vol
+from zcc import ControlPointDiscoveryService, ControlPointError
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import DOMAIN, MAC, TIMEOUT, VERBOSITY, WATCHDOG
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_HOST, default=""): str,
+        vol.Optional(CONF_PORT, default=5003): int,
+        vol.Required(MAC): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for zcc."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> config_entries.ConfigFlowResult:
+        """Handle the initial step."""
+        if user_input is None:
```

Comment:
personal preference invert: the flow (is not None)

Reply:
My logic is that if there is no data provided then fail straight away

Reply:
You need to use show for twice, whereas if you invert the flow you can usually do with one

Reply:
Refactored with inline, better descriptions, inverted flow.   See https://github.com/home-assistant/core/commit/7998adc1c0be9af723bc19f92660a8976a33db7a